sm_timing/src/Timer.cpp

```diff
@@ -237,6 +237,72 @@ namespace timing {
       out << std::endl;
     }
   }
+
```

Comment:
Ok removed duplication and added complexity, as usual :(. Still think this is better maintainable. What do you think @uschwes ?